sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala

```diff
@@ -408,8 +415,16 @@ object ExecutionPlanCaptureCallback {
       containsPlan(p.child, className)
     case p: ReusedExchangeExec =>
       containsPlan(p.child, className)
-    case p =>
-      p.expressions.exists(containsExpression(_, className))
+    case p if p.expressions.exists(containsExpression(_, className)) =>
+      true
+    case p: SparkPlan =>
+      regexMap.getOrElseUpdate(className,
+        Try(className.r).recoverWith { case t: Throwable =>
+          logWarning(s"Regex broken in a simpleStringWithNodeId match, $t")
```

Comment:
Why make this a log rather than actually throw?  If the regex is broken we want the test to fail so the regex can be fixed.  Logging a message in a test is very likely to be completely ignored, as most developers aren't scrubbing the test logs.

Reply:
This API does not handle regexes exclusively (and I did not want to create a new one).

Reply:
And what if the test passes when it shouldn't, because the regex was broken and didn't find what it was supposed to that would have caused the test to properly fail?  I'm really not a fan of this.